src/Algebra/Properties/Magma.agda

```diff
@@ -0,0 +1,74 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some items and theory for Magma
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra using (Magma)
+open import Algebra.Definitions using (Commutative)
+open import Data.Product using (_×_; _,_; ∃)
+open import Function using (_∘_)
+open import Level using (_⊔_)
+open import Relation.Binary using (Rel; _Respects_; _Respects₂_)
+open import Relation.Nullary using (¬_)
+
+module Algebra.Properties.Magma {a ℓ} (M : Magma a ℓ) where
+
+open Magma M
+
+------------------------------------------------------------------------------
+-- The divisibility relation
+------------------------------------------------------------------------------
+
+infix 5 _∣_ _∤_ _∣∣_ _¬∣∣_
+
+
+_∣_ : Rel Carrier (a ⊔ ℓ)          -- x ∣ y  denotes  x divides y
+x ∣ y = ∃ (\q → y ≈ (q ∙ x))
+
+_∤_ : Rel Carrier (a ⊔ ℓ)
+_∤_ x = ¬_ ∘ _∣_ x
```

Comment:
Much more readable as `x ∤ y = ¬ x ∣ y`. Same for `_¬∣∣_` below.

Reply:
Done.